src/library/scala/concurrent/impl/ExecutionContextImpl.scala

```diff
@@ -96,27 +102,22 @@ private[concurrent] object ExecutionContextImpl {
       case other => other.toInt
     }
 
-    def range(floor: Int, desired: Int, ceiling: Int) = scala.math.min(scala.math.max(floor, desired), ceiling)
-    val numThreads = getInt("scala.concurrent.context.numThreads", "x1")
-    // The hard limit on the number of active threads that the thread factory will produce
-    // scala/bug#8955 Deadlocks can happen if maxNoOfThreads is too low, although we're currently not sure
-    //         about what the exact threshold is. numThreads + 256 is conservatively high.
-    val maxNoOfThreads = getInt("scala.concurrent.context.maxThreads", "x1")
+    def range(floor: Int, desired: Int, ceiling: Int): Int = scala.math.min(scala.math.max(floor, desired), ceiling)
 
     val desiredParallelism = range(
       getInt("scala.concurrent.context.minThreads", "1"),
-      numThreads,
-      maxNoOfThreads)
+      getInt("scala.concurrent.context.numThreads", "x1"),
```

Comment:
what does it mean "x1"?

Reply:
please, ignore this. I have found what it means.

Reply:
:+1: